### ISSUE TITLE
Bitcoin_AI_Project

### ISSUE THREAD

**Issue by skmalman9ouri** [freeall/create-repository#855]

**Comment by skmalman9ouri** [comment]
import pandas as pd
import numpy as np
import talib
from sklearn.model_selection import train_test_split
from sklearn.ensemble import RandomForestClassifier
from sklearn.preprocessing import StandardScaler
import matplotlib.pyplot as plt

# 🔥 1. إنشاء بيانات سعر البيتكوين الوهمية
np.random.seed(42)
dates = pd.date_range(start="2023-01-01", periods=500, freq="D")
prices = np.cumsum(np.random.randn(500) * 50) + 25000  # بيانات وهمية لسعر البيتكوين

# إنشاء DataFrame
df = pd.DataFrame({"Date": dates, "Close": prices})
df.set_index("Date", inplace=True)

# 🔥 2. حساب المؤشرات الفنية
df["SMA"] = talib.SMA(df["Close"], timeperiod=14)  # المتوسط المتحرك البسيط
df["EMA"] = talib.EMA(df["Close"], timeperiod=14)  # المتوسط المتحرك الأسي
df["RSI"] = talib.RSI(df["Close"], timeperiod=14)  # مؤشر القوة النسبية
df["MACD"], df["MACD_signal"], _ = talib.MACD(df["Close"], fastperiod=12, slowperiod=26, signalperiod=9)

# 🔥 3. تحديد إشارات البيع والشراء
def signal(row):
    if row["RSI"] < 30 and row["MACD"] > row["MACD_signal"]:
        return "شراء 📈"
    elif row["RSI"] > 70 and row["MACD"] < row["MACD_signal"]:
        return "بيع 📉"
    else:
        return "انتظار ⏳"

df["Signal"] = df.apply(signal, axis=1)

# 🔥 4. تجهيز البيانات لنموذج الذكاء الاصطناعي
df.dropna(inplace=True)
X = df[["SMA", "EMA", "RSI", "MACD", "MACD_signal"]].values
y = np.where(df["Close"].shift(-1) > df["Close"], 1, 0)  # 1 إذا السعر سيرتفع، 0 إذا سينخفض

# 🔥 5. تقسيم البيانات إلى تدريب واختبار
X_train, X_test, y_train, y_test = train_test_split(X, y, test_size=0.2, random_state=42)

# 🔥 6. تطبيع البيانات لتحسين دقة النموذج
scaler = StandardScaler()
X_train = scaler.fit_transform(X_train)
X_test = scaler.transform(X_test)

# 🔥 7. إنشاء نموذج الذكاء الاصطناعي وتدريبه
model = RandomForestClassifier(n_estimators=100, random_state=42)
model.fit(X_train, y_train)

# 🔥 8. تقييم النموذج
accuracy = model.score(X_test, y_test)
print(f"🎯 دقة النموذج: {accuracy:.2%}")

# 🔥 9. تنبؤ السعر المستقبلي
next_prediction = model.predict([X_test[-1]])
prediction_result = "🔼 ارتفاع 📈" if next_prediction[0] == 1 else "🔽 انخفاض 📉"
print("🔮 التوقع التالي:", prediction_result)

# 🔥 10. رسم البيانات الفعلية والتوقعات
plt.figure(figsize=(12, 6))
plt.plot(df.index, df["Close"], label="السعر الفعلي", color="blue")
plt.title("تحليل سعر البيتكوين باستخدام الذكاء الاصطناعي")
plt.xlabel("التاريخ")
plt.ylabel("السعر")
plt.legend()
plt.show()
